pkg/image/prune/controller.go

```diff
@@ -0,0 +1,185 @@
+package prune
+
+import (
+	"fmt"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/golang/glog"
+	gonum "github.com/gonum/graph"
+	"github.com/openshift/origin/pkg/api/graph"
+	"github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+type PruningController struct {
+	images                 client.ImageInterface
+	streams                client.ImageStreamInterface
+	pods                   kclient.PodInterface
+	replicationControllers kclient.ReplicationControllerInterface
+	registryURL            string
+}
+
+func (c *PruningController) Run() error {
+	images, err := c.images.List(labels.Everything(), fields.Everything())
+	if err != nil {
+		return fmt.Errorf("error listing all images: %v", err)
+	}
+
+	streams, err := c.streams.List(labels.Everything(), fields.Everything())
+	if err != nil {
+		return fmt.Errorf("error listing all image streams: %v", err)
+	}
+
+	pods, err := c.pods.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error listing all pods: %v", err)
+	}
+
+	rcs, err := c.replicationControllers.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error listing all replication controllers: %v", err)
+	}
+
+	g := graph.New()
+	c.addImagesToGraph(g, images)
+	c.addImageStreamsToGraph(g, streams)
+	c.addPodsToGraph(g, pods)
+	c.addReplicationControllersToGraph(g, rcs)
+	c.prune(g)
+
+	return nil
+}
+
+const minimumImageAgeInMinutesToPrune = 60
+
+func (c *PruningController) addImagesToGraph(g graph.Graph, images *imageapi.ImageList) {
```

Comment:
so would you do this?

```go
type ImagePruner struct {
  g graph.Graph
  images client.ImageInterface
  // ...
}
```

And what would the func that takes an Image that you pass to Run do?